File: java/src/main/java/vsdk/toolkit/common/statistics/ShadowStatistics.java
package vsdk.toolkit.common.statistics;

public class ShadowStatistics {
    public int numberOfShadowRays;
    public int numberOfShadowCacheHits;

    public ShadowStatistics() {
        numberOfShadowRays = 0;
        numberOfShadowCacheHits = 0;
    }

    public void reset() {
        numberOfShadowRays = 0;
        numberOfShadowCacheHits = 0;
    }
}

File: typescript/src/main/vsdk/toolkit/common/statistics/ShadowStatistics.ts
export class ShadowStatistics {
  public numberOfShadowRays: number;
  public numberOfShadowCacheHits: number;

  public constructor() {
    this.numberOfShadowRays = 0;
    this.numberOfShadowCacheHits = 0;
  }

  public reset(): void {
    this.numberOfShadowRays = 0;
    this.numberOfShadowCacheHits = 0;
  }
}
